test/lib/auth.js

```diff
@@ -63,6 +69,179 @@ describe('Auth lib', function () {
         };
     };
 
+    describe('Update party API', function () {
+	
+	var auth = getAuthLib(strategy, {});
+	var req = {
+	    id: 'test_user',
+	    appId: config.oauth2.clientID,
+	    user: {accessToken: 'token',
+		   id: 'rick',
+		   nickName: 'theMagician',
+		   organizations: [{
+		       id: '123456789',
+		       name: 'patty',
+		       roles: [{
+			   id: '3',
+			   name: 'Seller'
+		       }, {
+			   id: '6',
+			   name: 'provider'
+		       }]
+		   }, {
+		       id: '987654321',
+		       name: 'MntyPythn',
+		       roles: [{
+			   id: '6',
+			   name: 'provider'
+		       }]
+		   }, {
+		       id: '111555999',
+		       name: 'AmaneceQueNoEsPoco',
+		       roles: [{
+			   id: '3',
+			   name: 'Seller'
+			}, {
+			    id: '1',
+			    name: 'purchuaser'
+			}]
+		   }]
+		  }
+        };
+	
+	it ('should continue with middleware processing if not user object has been provided', function (done) {
+	    var req = {
+		id: 'test_user',
+		appId: config.oauth2.clientID,
+		accesstoken: 'token'
+            };
+	    auth.checkOrganizations(req, {}, function () {
+		done();
+	    });
+	});
+
+	it ('should continue with middleware processing if the request have been already processed', function (done) {
+	    var req = {
+		id: 'test_user',
+		appId: config.oauth2.clientID,
+		user: { accessToken: 'token',
+			id: 'eugenio'}
+            };
+	    auth.getCache()['token'] = {orgProcessed: true};
+	    auth.checkOrganizations(req, {}, done);
+	});
+
+	it ('should continue with middleware processing if the request if currently being processed', function (done) {
+	    var req = {
+		id: 'test_user',
+		appId: config.oauth2.clientID,
+		user: { accessToken: 'token',
+			id: 'eugenio'}
+            };
+	    auth.getCache()['token'] = { orgProcessed : false, orgProcessing: true};
+	    auth.checkOrganizations(req, {}, function () {
+		done();
+	    });
+	});
+
+	it ('should continue with middleware processing if getOrganization call fails', function (done) {
+	    var getOrganizations = function (callback){
+		return callback({status: 500, message: 'An error occurred during request', body: 'Server error'});
+	    };
+	    var auth = getAuthLib(strategy, {}, null, getOrganizations);
+
+	    auth.getCache()['token'] = { orgProcessed : false, orgProcessing: false};
+	    auth.checkOrganizations(req, {}, function (){
```

Comment:
You should validate that getOrganization has been called, otherwise you cannot ensure that next has been called due to an error retrieving the organizations